Migrate Button, Radio, and DropdownMenu to ShadCN and Add Tests +3 issues resolved

idanidan29 opened the pull request:
**What kind of change does this PR introduce?**

- Adapted 3 components to use the ShadCN library.
- Created and adjusted Cypress test files for these 3 components and for the shadCN card element to ensure reliability and behavior coverage.
- While adapting the components, utilized ShadCN's `Button` ,`Radio`  and `Collapsible` elements.
- Added hover effects to the sections in the tools filter.
- Added a counter to each section indicating the number of active filters in use.

**Issue Number:**
- Closes #1656 
- Closes #1724
- Closes #1447 
- Closes #1591 


**Screenshots/videos:**

https://github.com/user-attachments/assets/2de706bd-673d-49ea-9025-3a414f07eecf



> The Radio component was not being used at the moment but still have been adapted to ShadCN for future use.

**Summary**

This PR:
- Migrated the existing `Button.tsx`, `Radio.tsx`, and `DropdownMenu` components to the ShadCN library.
- Enhanced the styling of the `button` component to improve dark mode compatibility (open to feedback).
- Added Cypress tests for all implemented components to ensure reliability and behavior coverage.
- solved 3 issues as related to phase 1:
  - Hover effects for better interactivity.
  - A visual counter showing the number of active filters per section.
  - fixed the checkbox icon to the middle

**Does this PR introduce a breaking change?**

no

# Checklist

Please ensure the following tasks are completed before submitting this pull request.

- [x] Read, understood, and followed the [contributing guidelines](https://github.com/json-schema-org/website/blob/main/CONTRIBUTING.md).

benjagm reviewed (changes requested):
Great work Idan. I just found some weird behaviours in the clear filters button. We need to review the logic to make sure whe are clearing all the field selections after clicking the clear filter button.

idanidan29 commented:
I've adjusted the clean filters logic, fixed the issue mentioned in [#1724](https://github.com/json-schema-org/website/issues/1724), and added tests to ensure correct sizing and placement of the checkbox icon
